types/tipset_test.go

```diff
@@ -54,220 +49,138 @@ func block(t *testing.T, height int, parentCid cid.Cid, parentWeight uint64, msg
-	pW := uint64(1337000)
-
-	b1 := block(t, 1, cid1, pW, "1")
-	b1.Ticket = []byte{0}
-	b2 := block(t, 1, cid1, pW, "2")
-	b2.Ticket = []byte{1}
-	b3 := block(t, 1, cid1, pW, "3")
-	b3.Ticket = []byte{0}
-	return b1, b2, b3
-}
-
-func RequireTestTipSet(t *testing.T) TipSet {
-	b1, b2, b3 := RequireTestBlocks(t)
-	return RequireNewTipSet(t, b1, b2, b3)
-}
-
-func TestTipSetAddBlock(t *testing.T) {
-	tf.UnitTest(t)
-
-	b1, b2, b3 := RequireTestBlocks(t)
-
-	// Add Valid
-	ts1 := TipSet{}
-	requireTipSetAdd(t, b1, ts1)
-	assert.Equal(t, 1, len(ts1))
-	requireTipSetAdd(t, b2, ts1)
-	requireTipSetAdd(t, b3, ts1)
-
-	ts2 := RequireNewTipSet(t, b1, b2, b3)
-	assert.Equal(t, ts2, ts1)
-
-	// Invalid height
-	b2.Height = 5
-	ts := TipSet{}
-	requireTipSetAdd(t, b1, ts)
-	err := ts.AddBlock(b2)
-	assert.Error(t, err)
-	b2.Height = b1.Height
-
-	// Invalid parent set
-	b2.Parents = NewSortedCidSet(cid1, cid2)
-	ts = TipSet{}
-	requireTipSetAdd(t, b1, ts)
-	err = ts.AddBlock(b2)
-	assert.Error(t, err)
-	b2.Parents = b1.Parents
-
-	// Invalid weight
-	b2.ParentWeight = Uint64(3000)
-	ts = TipSet{}
-	requireTipSetAdd(t, b1, ts)
-	err = ts.AddBlock(b2)
-	assert.Error(t, err)
-}
-
-func TestNewTipSet(t *testing.T) {
-	tf.UnitTest(t)
-
-	b1, b2, b3 := RequireTestBlocks(t)
-
-	// Valid blocks
-	ts, err := NewTipSet(b1, b2, b3)
-	assert.NoError(t, err)
-	assert.Equal(t, ts[b1.Cid()], b1)
-	assert.Equal(t, ts[b2.Cid()], b2)
-	assert.Equal(t, ts[b3.Cid()], b3)
-	assert.Equal(t, 3, len(ts))
-
-	// Invalid heights
-	b1, b2, b3 = RequireTestBlocks(t)
-	b1.Height = 3
-	ts, err = NewTipSet(b1, b2, b3)
-	assert.Error(t, err)
-	assert.Nil(t, ts)
-	b1.Height = b2.Height
-
-	// Invalid parent sets
-	b1, b2, b3 = RequireTestBlocks(t)
-	b1.Parents = NewSortedCidSet(cid1, cid2)
-	ts, err = NewTipSet(b1, b2, b3)
-	assert.Error(t, err)
-	assert.Nil(t, ts)
-	b1.Parents = b2.Parents
-
-	// Invalid parent weights
-	b1, b2, b3 = RequireTestBlocks(t)
-	b1.ParentWeight = Uint64(3000)
-	ts, err = NewTipSet(b1, b2, b3)
-	assert.Error(t, err)
-	assert.Nil(t, ts)
-}
-
-func TestTipSetMinTicket(t *testing.T) {
-	tf.UnitTest(t)
-
-	ts := RequireTestTipSet(t)
-	mt, err := ts.MinTicket()
-	assert.NoError(t, err)
-	assert.Equal(t, Signature([]byte{0}), mt)
-}
-
-func TestTipSetHeight(t *testing.T) {
-	tf.UnitTest(t)
-
-	ts := RequireTestTipSet(t)
-	h, err := ts.Height()
-	assert.NoError(t, err)
-	assert.Equal(t, uint64(43), h)
-}
-
-func TestTipSetParents(t *testing.T) {
-	tf.UnitTest(t)
-
-	b1, _, _ := RequireTestBlocks(t)
-	ts := RequireTestTipSet(t)
-	ps, err := ts.Parents()
-	assert.NoError(t, err)
-	assert.Equal(t, ps, b1.Parents)
-}
-
-func TestTipSetParentWeight(t *testing.T) {
-	tf.UnitTest(t)
-
-	ts := RequireTestTipSet(t)
-	w, err := ts.ParentWeight()
-	assert.NoError(t, err)
-	assert.Equal(t, w, uint64(1337000))
-}
+	b1, b2, b3 := makeTestBlocks(t)
+
+	assert.False(t, NoTipSet.Defined())
+
+	t.Run("construction of singleton", func(t *testing.T) {
```

Comment:
There's nothing inherently different about multiple block and single block tipsets (other than single block tipsets don't exercise some code). We have too many places where we treat these differently (only because we can optimize the 1 block case) already. This test should just test IsSolo (if you choose to keep it).

Len, ParentWeight, ToSortedCidSet, Parents, Height, and MinTicket should have separate test cases for documentation purposes and to be more explicit about what's under test.